<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>835</width>
    <height>698</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QR Encoder</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QRrenderer" name="QRRenderArea">
        <property name="minimumSize">
         <size>
          <width>755</width>
          <height>375</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>755</width>
          <height>375</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>17</height>
         </size>
        </property>
        <property name="text">
         <string>Text</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="textToEncode">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>620</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Shrinking factor</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="redimQR">
        <property name="minimum">
         <number>20</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBox">
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QPushButton" name="generateQRButton">
        <property name="text">
         <string>Generate QR</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Save QR Code</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Advanced options</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>ECC                  </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="ECCComboBox">
           <property name="currentText">
            <string>Low</string>
           </property>
           <item>
            <property name="text">
             <string>Low</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Medium</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Quartile</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>High</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>158</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Compression</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="QualitySpinBox">
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>835</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QRrenderer</class>
   <extends>QGraphicsView</extends>
   <header>qrrenderer.hpp</header>
   <slots>
    <slot>redimensionQR(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>generateQRButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generateQR()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>547</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveQR()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>762</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>560</x>
     <y>548</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redimQR</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QRRenderArea</receiver>
   <slot>redimensionQR(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redimQR</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>762</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>redimQR</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>762</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>589</x>
     <y>440</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redimQR</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>487</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>762</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>generateQR()</slot>
  <slot>saveQR()</slot>
 </slots>
</ui>
